### ISSUE TITLE
Rethink Variadic arguments as optional parameters

### ISSUE THREAD

**Issue by kardianos** [moov-io/x12#19]
Right now in the file pkg, a number of constructor functions take a variadic arguments in place of an optional argument. I would recommend against such a practice; it isn't clear what the string array is for at first glance. I might suggest something like:
```go
package file
type Option struct {
    Terminator string
}
// Option may be omitted.
func NewFile(InterchangeRule, opt *Option) {...}
```

Another method to capture this might be to create a Constructor type, with perhaps a default package level var constructor, where new files and methods are created from, with the options on them. Or maybe not.

**Comment by adamdecaf** [comment]
I agree. We should use `struct`s to modify behavior. cc @mfdeveloper508
